src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotManager.java

```diff
@@ -150,14 +168,29 @@ protected void purgeAndCommit() {
     }
   }
 
+  /**
+   * If a previous realtime update has changed which trip pattern is associated with the given trip
+   * on the given service date, this method will dissociate the trip from that pattern and remove
+   * the trip's timetables from that pattern on that particular service date.
+   *
+   * For this service date, the trip will revert to its original trip pattern from the scheduled
+   * data, remaining on that pattern unless it's changed again by a future realtime update.
+   *
+   * @return true if the trip was found to be shifted to a different trip pattern by a realtime
+   * message and an attempt was made to re-associate it with its originally scheduled trip pattern.
+   */
+  public void revertTripToScheduledTripPattern(FeedScopedId tripId, LocalDate serviceDate) {
+    buffer.revertTripToScheduledTripPattern(tripId, serviceDate);
+  }
+
   /**
    * Remove realtime data from previous service dates from the snapshot. This is useful so that
    * instances that run for multiple days don't accumulate a lot of realtime data for past
    * dates which would increase memory consumption.
    * If your OTP instances are restarted throughout the day, this is less useful and can be
    * turned off.
    */
-  protected final boolean purgeExpiredData() {
+  private boolean purgeExpiredData() {
```

Comment:
Could you document the return value of this method as well as the return value of the corresponding method in TimetableSnapshot?